runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -256,34 +251,10 @@ void registerTable(@Nonnull String namespace,
                 if (oldRecord == null) { // Case 1 above, new unseen table
                     this.registryTable.put(tableNameKey,
                         new CorfuRecord<>(tableDescriptors, metadataBuilder.build()));
-                } else {
-                    // Case 2 above, re-open of existing table
-                    // If the schema has changed and/or there is an upgrade
-                    // from a previous version which did not contain the Any
-                    // fields for key/value/metadata, we need to update the
-                    // existing record in registry table.
-                    boolean schemaChanged = false;
-                    boolean upgradeCase = false;
-
-                    if (!oldRecord.getPayload().getFileDescriptorsMap().keySet()
-                        .equals(tableDescriptors.getFileDescriptorsMap().keySet())) {
-                        log.warn("Protobuf file list changed for {} table {}. Prev list {}, new list {}",
-                            namespace, tableName, oldRecord.getPayload().getFileDescriptorsMap().keySet(),
-                            tableDescriptors.getFileDescriptorsMap().keySet());
-                        schemaChanged = true;
-                    }
-
-                    if (!oldRecord.getPayload().hasKey()) {
-                        upgradeCase = true;
-                    }
-
-                    if (schemaChanged || upgradeCase) {
-                        this.registryTable.put(tableNameKey,
-                            new CorfuRecord<>(tableDescriptors, metadataBuilder.build()));
-                    } // else no need to re-add the exact same entry into the table
+
```

Comment:
Can you fail here ?

When would you have to calls that try to register the same table with different `tableDescriptors` ?

Reply:
We are not able to fail here right now, as the issue we observed will cause a false unequal.

Existing format is like `5: "messaging"` -> field name is replaced by the field number 
Incoming format is like `stream_tag: "messaging"`

So I added a trace level log here as a temporary solution.

Reply:
If you ignore upgrade - which this patch doesnt support - when does this case happen ? Why is the same table being opened with different options ? these options are persisted, so it doesnt make sense to have one copy of the table with multiple options, but only persist one set of options.

Reply:
@Maithem  Same table are opened with the same options, the difference is how they get encoded.

Essentially they represent the same schema, but right now `equals` does not treat them same. So Trying to skip the check and we can fail here once we figure out how to detect them as the same schema

Reply:
@zhangn49, how did this encoding change?

Reply:
@pankti-m it gets serialized once written, so then the serialized form (the one persisted) is compared to the non-serialized form (the one being freshly read from the proto file), and the equals "fails" to detect they are the same, only that one is serialized, the other isn't.

Reply:
Are you saying that the value being read back is failing the equality check because its a different references?


If the answer is yes, then it still doesn't make sense. The following code does a value comparison

```java

oldRecord.getPayload().getFileDescriptorsMap().keySet()
                        .equals(tableDescriptors.getFileDescriptorsMap().keySet()
```

since the equals method is overloaded in `FileDescriptorProto` and it actually does a value comparison, so even if the references are different it would return true if the nested values are equal.

Reply:
@Maithem 
No, they are different values - like "5" vs "stream_tag"

5 is the field number of "stream_tag"

message {
    stream_tag = 5;
}

So they are different values and forms.

Reply:
Ok, I get the problem now.

I still think silently omitting the case is problematic and can introduce subtle issues when a new schema is being registered, but omitted because an old one already exists. It should fail, but it doesn't and now the application can start up and use an old schema where they think that the new schema is being used (because the "Registration passes"). 

Omitting the second registration implicitly assume that the second registration will equal the first schema being registered. I think we should fail explicitly. The comparison can be achieved by normalizing both forms. You can serialize/deserialzie both old/new and check for equality. This is not efficient, but its also not on a hot path.

It's better to fail explicitly than assume it wont happen.